src/main/java/edu/ithaca/dragon/bank/BankAccount.java

```diff
@@ -30,21 +39,78 @@ public String getEmail(){
      * @post reduces the balance by amount if amount is non-negative and smaller than balance
      */
     public void withdraw (double amount) throws InsufficientFundsException{
+        if  (!isAmountValid(amount)){
+            throw new InsufficientFundsException("Not enough money");
+        }
+        if (amount <= 0){
+            throw new InsufficientFundsException("Not enough money");
+        }
         if (amount <= balance){
             balance -= amount;
         }
+        
         else {
             throw new InsufficientFundsException("Not enough money");
         }
     }
-
+    //Increases the balance by amount if amount is nonegative and does not have fraction of cents
+    public void deposit (double amount) throws IllegalArgumentException{
+        if  (!isAmountValid(amount)){
+            throw new IllegalArgumentException("invalid amount");
+        }
+        else{
+            balance +=amount;
+        }
+    }
 
     public static boolean isEmailValid(String email){
+        //Check if @ exits
         if (email.indexOf('@') == -1){
             return false;
         }
+        //Check if . extists
+        if (email.indexOf('.') == -1){
+            return false;
+        }
+        //Cant have double dash
+        if (email.indexOf("--") != -1){
+            return false;
+        }
+        //check for two terms after .
+        if (email.length()-email.indexOf('.')-1 <= 1){
```

Comment:
I detect that this code is problematic. According to the [Correctness (CORRECTNESS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [SA: Self comparison of value with itself (SA_SELF_COMPARISON)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#sa-self-comparison-of-field-with-itself-sa-field-self-comparison).
This method compares a field with itself, and may indicate a typo or a logic error. Make sure that you are comparing the right things.